tests/schemas/metadata/discovery_metadata.json

```diff
@@ -0,0 +1,81 @@
+{
+  "version": "1.5.2",
```

Comment:
In addition to this schema-only test (it's just the input!), it would be good to add in a dynamic test as well of the output generated_config -- see here:

```
peringknife@peringknife:~/udmi/tests/sites/basic/devices/AHU-1$ ls -l expected/generated_config.json 
-rw-r--r-- 1 peringknife primarygroup 651 Nov 20 22:12 expected/generated_config.json
```

Basically if you add something into that metadata.json file the test infrastructure picks it up and should generate (and then verify) the generated_config.json file.

You can run this test locally with something like "bin/test_sites basic"

You'll actually see that the current tests are failing with your code because of the comment below about "this should be null if the input is null" -- https://github.com/faucetsdn/udmi/actions/runs/11971809492/job/33377389955?pr=1033#step:12:62

One thing maybe to do next is to setup GitHub Actions on your own account so you can see the results locally!